lib/sugar_utils/file.rb

```diff
@@ -191,6 +192,63 @@ def self.write(filename, data, options = {}) # rubocop:disable MethodLength, Abc
       raise(Error, "Unable to write #{filename} with #{e}")
     end
 
+    # Atomically write to an existing file, overwriting it, or create the file
+    # if it does not exist.
+    #
+    # @note Either option :mode or :perm can be used to specific the permissions
+    # on the file being written to. This aliasing is used because both these
+    # names are used in the standard library, File.open uses :perm and FileUtils
+    # uses :mode. The user can choose whichever alias makes their code most
+    # readable.
+    #
+    # @param filename [String]
+    # @param data [#to_s]
+    # @param options [Hash]
+    # @option options [Integer] :timeout (10)
+    # @option options [Boolean] :flush (false)
+    # @option options [String, Integer] :owner
+    # @option options [String, Integer] :group
+    # @option options [Integer] :mode (0o644)
+    # @option options [Integer] :perm (0o644)
+    #
+    # @raise [SugarUtils::File::Error]
+    #
+    # @return [void]
+    def self.atomic_write(filename, data, options = {}) # rubocop:disable MethodLength, AbcSize
+      write_options = WriteOptions.new(filename, options)
+
+      # @see https://apidock.com/rails/File/atomic_write/class
```

Comment:
just curious why is this mentioned here? maybe adding some comments here comparing this implementation to the rails one would be nice.

Reply:
I can add that. Part of our reasoning was also to avoid including the whole of ActiveSupport; I will comment on that as well.